<!DOCTYPE html>
<html>
<head>
  <title>Programming Exercise</title>
  <style>
    p {
        font-size: 18px
    }
    pre {
        font-size: 16px
    }
    .long {
        overflow-wrap: break-word;
    }    
  </style>
</head>


<body>
  <h1>align2</h1>
  
  <h2>Description</h2>
  <p>Do a pairwise alignment of two sequences from a <a href="ppar.fasta">.FASTA</a> file containing multiple homologous sequences.
    The sequences to align are indicated by <b>Uniprot name</b>, which appears as part of the sequence id.
 
</p>

<!--
<h2>Hint</h2>
You can get any substitution matrix from Biopython

<pre><code>from Bio.Align import substitution_matrices
subst_mat = substitution_matrices.load("BLOSUM62")

</code></pre>
-->

<h2>Tests</h2>
  <pre>    >>> align2("ppar.fasta", "PPARD_XENLA", "PPARG_CANLF", -4)[0]
    876.0
    >>> align2("ppar.fasta", "AAAA_AAAA", "PPARG_CANLF", -4)[0]
    0
    </pre>

  <hr>

  <br>
<!--
  <h3>Related exercises:</h3>
  <p><a href="nw2.html">nw2</a></p>
  -->
</body>
</html>


